blenderproc/python/loader/URDFLoader.py

```diff
@@ -0,0 +1,454 @@
+from typing import List, Union
+import os
+from mathutils import Matrix, Vector
+import numpy as np
+
+import bpy
+
+from blenderproc.python.utility.SetupUtility import SetupUtility
+from blenderproc.python.utility.BlenderUtility import get_all_materials
+from blenderproc.python.utility.Utility import Utility
+from blenderproc.python.loader.ObjectLoader import load_obj
+from blenderproc.python.material import MaterialLoaderUtility
+from blenderproc.python.types.MaterialUtility import Material
+from blenderproc.python.types.MeshObjectUtility import MeshObject, create_primitive
+from blenderproc.python.types.URDFUtility import URDFObject
+from blenderproc.python.types.LinkUtility import Link
+from blenderproc.python.types.InertialUtility import Inertial
+from blenderproc.python.filter.Filter import one_by_attr
+from blenderproc.python.types.MeshObjectUtility import create_with_empty_mesh
+from blenderproc.python.types.BoneUtility import set_location_constraint, set_rotation_constraint, \
+    set_copy_rotation_constraint
+
+
+def load_urdf(urdf_file: str, weight_distribution: str = 'rigid',
+              fk_offset: Union[List[float], Vector, np.array] = [0., -1., 0.],
+              ik_offset: Union[List[float], Vector, np.array] = [0., 1., 0.]) -> URDFObject:
+    """ Loads a urdf object from an URDF file.
+
+    :param urdf_file: Path to the URDF file.
+    :param weight_distribution: One of ['envelope', 'automatic', 'rigid']. For more information please see
+                                https://docs.blender.org/manual/en/latest/animation/armatures/skinning/parenting.html.
+    :param fk_offset: Offset between fk (forward kinematic) bone chain and link bone chain. This does not have any effect on the
+                      transformations, but can be useful for visualization in blender.
+    :param ik_offset: Offset between ik (inverse kinematic) bone chain and link bone chain. Effects on the transformation (e.g.
+                      `urdf_object.set_location_ik()`) are being handled internally. Useful for visualization in
+                      blender.
+    :return: URDF object instance.
+    """
+    # install urdfpy
+    SetupUtility.setup_pip(user_required_packages=["git+https://github.com/wboerdijk/urdfpy.git"])
```

Comment:
Why do we need to use Wouts repo here?

Reply:
Its because the original urdfpy pip package does not support all urdfs in the way we want. Thats why wout adjusted a few things and made a custom version. But I think we already discussed once to move this repo to the DLR account eventually.